Libraries/libbom/Tools/lsbom.cpp

```diff
@@ -294,6 +294,81 @@ extern "C" {
     } __attribute__((packed));
 }
 
+static std::string
+PrintItemToString(Options::PrintItem item, std::string path, struct bom_path_info_2 *info)
+{
+    switch (item) {
+        case Options::PrintItem::Checksum:
+            if (info->type == bom_path_type_directory) {
+                return "";
+            }
+            return std::to_string(ntohl(info->checksum));
+        case Options::PrintItem::FileName:
+            return path;
+        case Options::PrintItem::FileNameQuoted:
+            return "\"" + path + "\"";
+        case Options::PrintItem::GroupID:
+            return std::to_string(ntohl(info->group));
+        case Options::PrintItem::GroupName:
+            return std::string(); // TODO: implement
+        case Options::PrintItem::Permissions: {
+            std::stringstream stream;
+            stream << std::oct << ntohs(info->mode);
+            return stream.str();
+            }
+        case Options::PrintItem::PermissionsText:
+            return std::string(); // TODO: implement
+        case Options::PrintItem::FileSize:
+            if (info->type == bom_path_type_directory) {
+                return "";
+            }
+            return std::to_string(ntohl(info->size));
+        case Options::PrintItem::FileSizeFormatted:
+            return std::string(); // TODO: implement
+        case Options::PrintItem::ModificationTime:
+            return std::to_string(ntohl(info->modtime));
+        case Options::PrintItem::ModificationTimeFormatted:
+            return std::string(); // TODO: implement
+        case Options::PrintItem::UserID:
+            return std::to_string(ntohl(info->user));
+        case Options::PrintItem::UserName:
+            return std::string(); // TODO: implement
+        case Options::PrintItem::UserGroupID:
+            return std::to_string(ntohl(info->user)) + "/" + std::to_string(ntohl(info->group));
+        case Options::PrintItem::UserGroupName:
+            return std::string(); // TODO: implement
+    }
+}
+
+static std::vector<Options::PrintItem>
+GetDefaultPrintFormat ()
+{
+    std::vector<Options::PrintItem> toReturn;
+
+    toReturn.push_back(Options::PrintItem::FileName);
+    toReturn.push_back(Options::PrintItem::Permissions);
+    toReturn.push_back(Options::PrintItem::UserGroupID);
+    toReturn.push_back(Options::PrintItem::FileSize);
+    toReturn.push_back(Options::PrintItem::Checksum);
+
+    return toReturn;
```

Comment:
This could use an "initializer list" to create the vector:

    std::vector<Options::PrintItem> toReturn = {
        Options::PrintItem::FileName,
        Options::PrintItem::Permissions,
        Options::PrintItem::UserGroupID,
        Options::PrintItem::FileSize,
        Options::PrintItem::Checksum,
    };

    return toReturn;

Or even remove the temporary variable completely:

    return {
        Options::PrintItem::FileName,
        Options::PrintItem::Permissions,
        Options::PrintItem::UserGroupID,
        Options::PrintItem::FileSize,
        Options::PrintItem::Checksum,
    };

I like that style better for two reasons. Most importantly, I think it's easier to read: the structure of the code matches the structure of the data structure created. And not important in this case, but it's also slightly faster since it avoids calling `push_back` five times, each of which might need to expand the vector's allocated space.